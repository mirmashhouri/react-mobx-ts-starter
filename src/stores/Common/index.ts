// Stores
export { LanguageStore, useLanguageStore } from './Language/store';
export { ThemeStore, useThemeStore } from './Theme/store';

