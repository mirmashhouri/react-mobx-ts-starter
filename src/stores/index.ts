// Stores

export * from './Common';
export * from './Config/store';
export * from './Global/store';
export * from './Global/Lazy/store';
export * from './Global/Core/store';

// Types
export * from './Config/types';
export * from './Global/types';
export * from './Global/Core/types';
export * from './Global/Lazy/types';

export * from './Config/initialValues';
