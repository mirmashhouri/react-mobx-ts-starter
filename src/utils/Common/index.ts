export * from './getLocale';
export * from './i18n';
export * from './lazyLoadImport';