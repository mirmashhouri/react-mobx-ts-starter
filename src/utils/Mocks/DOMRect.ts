
global.DOMRect = class DOMRect {
  bottom: number = 0;

  left: number = 0;

  right: number = 0;

  top: number = 0;

  constructor(public x = 0, public y = 0, public width = 0, public height = 0) {}

  static fromRect(other?: DOMRectInit): DOMRect {
    return other ? new DOMRect(other.x, other.y, other.width, other.height) : new DOMRect();
  }

  toJSON() {
    return JSON.stringify(this);
  }
};
