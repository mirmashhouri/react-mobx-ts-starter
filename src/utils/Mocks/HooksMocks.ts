import { useEffect } from 'react';
import {
  LanguageStore,
  ThemeStore,
  ConfigStore,
} from 'stores/index';
import 'intersection-observer';

jest.mock('utils/Hooks', () => ({
  useLanguageStore: () => LanguageStore,
  useThemeStore: () => ThemeStore,
  useConfigStore: () => ConfigStore,

  useLanguages: () => ({
    isOpen: true,
    attachListeners: () => {},
    close: () => {},
  }),

  useNavigateTo: () => ({
    navigate: () => {},
  }),

  useIntersectionObserver: () => ({
    observer: undefined,
  }),

  useIsomorphicLayoutEffect: () => useEffect(() => {}, []),

  useEventListener: () => {},

  useOnClickOutside: () => {},

  useOnLocationChange: () => {},
}));