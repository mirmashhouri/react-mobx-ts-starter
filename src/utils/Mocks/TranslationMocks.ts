jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => ({
    t: (str: string) => str,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
      exists: () => true,
    },
  }),
  initReactI18next: {
    type: '3rdParty',
    init: jest.fn(),
  },
}));