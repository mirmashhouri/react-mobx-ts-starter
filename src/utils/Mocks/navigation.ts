const mockResponse = jest.fn();
Object.defineProperty(window, 'location', {
  value: {
    hash: '',
    assign: mockResponse,
    replace: mockResponse,
    reload: mockResponse,
    pathname: '',
    href: '',
    search: '',
    origin: 'http://localhost',
    host: '',
    hostName: '',
    port: '',
    JSON: '',
    protocol: '',
  },
  writable: true,
});
