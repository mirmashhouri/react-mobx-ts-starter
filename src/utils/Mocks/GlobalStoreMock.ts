import { GlobalStore } from 'stores';

export const globalStoreMock = new GlobalStore();

Element.prototype.scrollTo = () => {};