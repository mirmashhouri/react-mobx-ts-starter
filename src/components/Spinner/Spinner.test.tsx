/* eslint-disable import/no-extraneous-dependencies */
import { render } from '@testing-library/react';

import { CoreStoreProvider } from 'stores';
import { globalStoreMock } from 'utils/Mocks';
require('utils/Mocks/HooksMocks');
require('utils/Mocks/navigation');
require('utils/Mocks/TranslationMocks');
import { Spinner } from './';

jest.mock('app/App', () => ({
  useAppContext: () => ({
    store: globalStoreMock,
  }),
}));

describe('Spinner component', () => {
  const imageElement = 'img';
  const altAttribute = 'alt';
  const altText = 'React-MobX-Starter';
  const spinnerClassName = 'spinner-border';

  const Component = (
    <CoreStoreProvider.Provider value={ globalStoreMock.coreStore }>
      <Spinner />
    </CoreStoreProvider.Provider>
  );

  it('Render the Image', () => {
    const renderer = render(Component);

    //Expect to render alt attribute
    expect(
      renderer.container.getElementsByTagName(imageElement)[0],
    ).toHaveAttribute(altAttribute, altText);
  });

  it('Render Spinner', () => {
    //Expect to render spinner
    expect(
      render(Component).container.getElementsByClassName(spinnerClassName),
    ).toHaveLength(1);
  });

  it('Render to match snapshot', () => {
    //Expect to match snapshot
    expect(
      render(Component).container,
    ).toMatchSnapshot();
  });
});