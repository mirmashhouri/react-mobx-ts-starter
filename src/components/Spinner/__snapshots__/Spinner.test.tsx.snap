// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component Render to match snapshot 1`] = `
<div>
  <div
    class="spinner-icon position-absolute top-50 left-50 w-10-rem h-10-rem text-center translate-middle z-index-layout"
  >
    <div
      class="position-absolute w-100 h-100 top-50 left-50 translate-middle"
    >
      <div
        class="spinner-border w-100 h-100 fs-1"
        role="status"
      />
    </div>
    <div
      class="position-absolute w-75 top-50 left-50 translate-middle p-2"
    >
      <img
        alt="React-MobX-Starter"
        src="http://localhost/assets/logo.png"
      />
    </div>
  </div>
</div>
`;
